collector/test/CollectorConfigTest.cpp

```diff
@@ -145,4 +151,54 @@ TEST(CollectorConfigTest, TestEnableExternalIpsRuntimeConfig) {
   EXPECT_TRUE(config.EnableExternalIPs());
 }
 
+TEST(CollectorConfigTest, TestConfigMapTrue) {
+  std::string jsonStr = R"({
+           "networkConnectionConfig": {
+             "enableExternalIps": true
```

Comment:
I also now realize the only place I can actually check what the configuration is going to look like is these tests, which is not great, we need a way to have a formal specification of the configuration that we can properly review, but for now I guess it's fine 🤷🏻‍♂️ .

I do have some objections though:
-  `networkConnectionConfig` is redundant, we know this is a configuration parameter, so it should be `networkConnection`.
- `enableExternalIps` is not extensible.

I would prefer the configuration to look something closer to this:
```yaml
networkConnection:
  externalIPs:
    enable: true
```

Reply:
If you want to change these, we need to change the protobuf definitions.

Reply:
Why not? Is not like the feature is GA.